src/compiler/checker.ts

```diff
@@ -2028,6 +2028,10 @@ namespace ts {
                         let suggestion: Symbol | undefined;
                         if (suggestedNameNotFoundMessage && suggestionCount < maximumSuggestionCount) {
                             suggestion = getSuggestedSymbolForNonexistentSymbol(originalLocation, name, meaning);
+                            const isGlobalScopeAugmentationDeclaration = suggestion && suggestion.valueDeclaration && isAmbientModule(suggestion.valueDeclaration) && isGlobalScopeAugmentation(suggestion.valueDeclaration);

```

Comment:
In hindsight I wish I'd named this `suggestionIs...`, but it's already obnoxiously long.

Maybe `suggestionIsGlobalDeclaration`? `suggestionIsGlobalModule`? Something else?

Reply:
in context, I think this is a pretty good name.